src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1851 @@ interface MPSMatrixMultiplication {
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:primaryImage:secondaryImage:destinationImage:")]
+		void Encode (IMTLCommandBuffer commandBuffer, MPSImage primaryImage, MPSImage secondaryImage, MPSImage destinationImage);
+
+		[Export ("encodeToCommandBuffer:primaryImage:secondaryImage:")]
+		MPSImage EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSImage primaryImage, MPSImage secondaryImage);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronPReLU")]
+	[DisableDefaultCtor]
+	interface MPSCnnNeuronPReLU {
+		[Export ("initWithDevice:a:count:")]
+		[Internal, Sealed]
+		IntPtr InitWith (IMTLDevice device, IntPtr /* float* */ a, nuint count);
+
+		// inlining ctor from base class
+		[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronHardSigmoid")]
+	interface MPSCnnNeuronHardSigmoid {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+
+		[Export ("initWithDevice:a:b:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a, float b);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronSoftPlus")]
+	interface MPSCnnNeuronSoftPlus {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+
+		[Export ("initWithDevice:a:b:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a, float b);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronSoftSign")]
+	interface MPSCnnNeuronSoftSign {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronELU")]
+	interface MPSCnnNeuronELU {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("initWithDevice:a:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronReLUN")]
+	interface MPSCnnNeuronReLUN {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+
+		[Export ("initWithDevice:a:b:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a, float b);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnConvolutionDescriptor), Name = "MPSCNNSubPixelConvolutionDescriptor")]
+	interface MPSCnnSubPixelConvolutionDescriptor {
```

Comment:
the base type has a DisableDefaultCtor so there needs to be a reason for not having it here